shakenfist/external_api/base.py

```diff
@@ -156,16 +181,36 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def get_network_uuid(network_name):
+    if network_name and valid_uuid4(network_name):
+        # Already a valid UUID
+        return network_name
+
+    # Check if valid name of an active network
+    filters = [
+        partial(baseobject.namespace_filter, get_jwt_identity()[0]),
+        baseobject.active_states_filter,
+    ]
+    for n in net.Networks(filters):
+        # Duplicate name behaviour is "not defined" for the user therefore we
+        # just return the first network found.
+        if n.name == network_name:
+            return n.uuid
+
+
 def arg_is_network_uuid(func):
     # Method uses the network from the db
     def wrapper(*args, **kwargs):
-        if 'network_uuid' in kwargs:
-            kwargs['network_from_db'] = Network.from_db(
-                kwargs['network_uuid'])
-        if not kwargs.get('network_from_db'):
-            LOG.with_field('network', kwargs['network_uuid']).info(
-                'Network not found, missing or deleted')
-            return error(404, 'network not found')
+        with ThreadLocalReadOnlyCache():
+            net_uuid = get_network_uuid(kwargs.get('network_uuid'))
```

Comment:
Why are network lookups a method, and instance lookups are inline? You could probably also write a more generic "object lookup" by providing an object type to lookup method mapping if that makes sense.

Reply:
The network lookup is also used at https://github.com/shakenfist/shakenfist/blob/647f797b6f4f564db4b89e14af28cdeb21d3c042/shakenfist/external_api/instance.py#L270

It's a good idea to make it more generic, since the concept might stretch to artifacts.